spec/models/filters/filter_by_repeatedness_spec.rb

```diff
@@ -41,18 +41,17 @@
     describe 'set to keep first' do
       it 'keeps the first classification for a given user' do
         extracts = [
-          Extract.new(id: 1, classification_id: 1, user_id: 1, extractor_key: "a"),
-          Extract.new(id: 2, classification_id: 1, user_id: 1, extractor_key: "b"),
-          Extract.new(id: 3, classification_id: 2, user_id: 2, extractor_key: "a"),
-          Extract.new(id: 4, classification_id: 2, user_id: 2, extractor_key: "b"),
-          Extract.new(id: 5, classification_id: 3, user_id: 1, extractor_key: "a"),
-          Extract.new(id: 6, classification_id: 3, user_id: 1, extractor_key: "b")
+          Extract.new(id: 1, classification_id: 1, user_id: 1, extractor_key: 'a', classification_at: (Time.now - 1.minute)),
+          Extract.new(id: 2, classification_id: 1, user_id: 1, extractor_key: 'b', classification_at: (Time.now - 1.minute)),
+          Extract.new(id: 3, classification_id: 2, user_id: 2, extractor_key: 'a', classification_at: (Time.now - 1.minute)),
+          Extract.new(id: 4, classification_id: 2, user_id: 2, extractor_key: 'b',  classification_at: (Time.now - 1.minute)),
+          Extract.new(id: 5, classification_id: 3, user_id: 1, extractor_key: 'a',  classification_at: (Time.now - 2.minute)),
```

Comment:
Correct me if I'm wrong, but I believe there is generally an expected ordering such that extract ID and classification_at increase with time.  Therefore, I recommend you change the classification_at for id 5 and 6 to Time.now (as you do below).

However, it might also be worth adding a second test where the same result is expected when the array of extracts is reversed (so we know that the sorting is working whether or not the input array happens to be ordered that way).

Reply:
Are the repeated `Time.now` calls when creating the array if unsaved extracts is supposed to represent time actually passing? Like if between creating extract ids 2 and 3, the second ticks over--that's intended, right? I typically feel like using Time can be brittle and save it to a variable first, assuming they're supposed to be the same. But if the intention here is to have time pass and that they're not supposed to match, maybe you could spread them out more instead of having the first four be `1.minute` ago.

Reply:
not my intention, maybe i dont fully understand the question, but I added Time.now calls on classification_at to test the sorting. 

we can view extracts with id 2 and 3 as extracts coming from two different users that classified at around the same time. but we can sparse time out if we want. no issue with that. we probably want extracts due to the same classification to have the same classification_at time though

Reply:
I'm just pointing out that for instance it is possible that extracts 1 and 2, generated from the same classification, could have different `classification_at` values in this spec since time is passing during their creation. 

I don't think that matters for this test specifically, but that's why I will generally store the Time to a variable and affect it in obvious ways rather than call `Time.now` multiple times.

Reply:
ah ok.  i see what youre getting at.